src/webui/api/torrentscontroller.cpp

```diff
@@ -559,12 +559,43 @@ void TorrentsController::filePrioAction()
     checkParams({"hash", "id", "priority"});
 
     const QString hash = params()["hash"];
-    int fileID = params()["id"].toInt();
-    int priority = params()["priority"].toInt();
+    bool ok = false;
+    const int priority = params()["priority"].toInt(&ok);
+    if (!ok)
+        throw APIError(APIErrorType::BadParams, tr("Priority must be an integer"));
+
+    if (!BitTorrent::FilePriority::isValidPriority(priority))
+        throw APIError(APIErrorType::BadParams, tr("Priority is not valid"));
+
     BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
+    if (!torrent)
+        throw APIError(APIErrorType::NotFound);
+    if (!torrent->hasMetadata())
+        throw APIError(APIErrorType::Conflict, tr("Torrent's metadata has not yet downloaded"));
+
+    const int filesCount = torrent->filesCount();
+    QList<int> fileIds;
```

Comment:
It's better to avoid using of QList unless it's strongly required. Use QVector instead. And reserve needed amount of items before fill it.

Reply:
Will be removed due to merging the two loops